### PR TITLE
pablo_lara

### PR DESCRIPTION
**Pull request by larabjj**
Exercise-01 k8s training

### CODE REVIEW

File: pablo_lara/session-01/exercise-01/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+I've used all files include the folder "exercise-01". Also, I've created a new files like *wordpress-secret.yaml* or *mariadb-secret.yaml* in order to pass sensitive information.
+
+## Commands.bash
+
+Before executing this command, I would recommend executing in the command line
+~~~
+kubectl delete ns exercise-01
+~~~
+to ensure that *namespace=exercise-01* it's not exist.
```
[the remark this answers is not in the document]
Comment (larabjj) [reply pablo_lara/session-01/exercise-01/README.md]: it's done! 3rd person, thanks.

File: pablo_lara/session-01/exercise-01/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    command:
+    - mysqladmin
+    - ping
+  initialDelaySeconds: 5
+  periodSeconds: 5
+~~~
+
+At last, I've configured *wordpress-canary-deployment.yaml* like *canary version with wordpress 4.9.8*
+
+Once an executed commands.bash, should see the port assigned to *wordpress-svc* type NodePort to open it in your web browser.
```
[the remark this answers is not in the document]
Comment (larabjj) [reply pablo_lara/session-01/exercise-01/README.md]: ## Thanks so much

File: pablo_lara/session-01/exercise-01/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+Before executing this command, I would recommend executing in the command line
+~~~
+kubectl delete ns exercise-01
+~~~
+to ensure that *namespace=exercise-01* it's not exist.
+
+## Liveness & Readiness Probe
+
+In the configuration file, you can see that the Deployment *wordpress-deployment.yaml* and *wordpress-canary-deployment.yaml* it has the same probe. An HTTP request to a specific route server, although I'm not sure that it's working fine.
```
[the remark this answers is not in the document]
Comment (larabjj) [reply pablo_lara/session-01/exercise-01/README.md]: Yes, sorry. The LP and RP seems to be fine.

File: pablo_lara/session-01/exercise-01/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+At last, I've configured *wordpress-canary-deployment.yaml* like *canary version with wordpress 4.9.8*
+
+Once an executed commands.bash, should see the port assigned to *wordpress-svc* type NodePort to open it in your web browser.
+~~~
+NAME            TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)        AGE
+mariadb         ClusterIP   10.105.9.227    <none>        3306/TCP       1s
+wordpress-svc   NodePort    10.104.36.239   <none>        80:30277/TCP   1s
+~~~
+~~~
+larabjj@ubuntu:~/Desktop/GitHub/pablo_lara/session-01/exercise-01$ kubectl get pods --namespace=exercise-01
```
[the remark this answers is not in the document]
Comment (larabjj) [reply pablo_lara/session-01/exercise-01/README.md]: Ok, it's done!

File: pablo_lara/session-01/exercise-01/commands.bash [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+## Create namespace
+kubectl create -f ns.yaml
+## Create secrets for mariadb and wordpress
+kubectl create -f mariadb-secret.yaml
+kubectl create -f wordpress-secret.yaml
+## Check secrets
```
[the remark this answers is not in the document]
Comment (larabjj) [reply pablo_lara/session-01/exercise-01/commands.bash]: It's done!

File: pablo_lara/session-01/exercise-01/wordpress-canary-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        ports:
+        - name: wordpress
+          containerPort: 80
+        env:
+          - name: WORDPRESS_DATABASE_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: wordpress-secret
+                key: mariadb-database-password
+          - name: WORDPRESS_DB_HOST
```
[the remark this answers is not in the document]
Comment (larabjj) [reply pablo_lara/session-01/exercise-01/wordpress-canary-deployment.yaml]: you're right ! this is not env. var

File: pablo_lara/session-01/exercise-01/wordpress-canary-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            valueFrom:
+              secretKeyRef:
+                name: wordpress-secret
+                key: password-wp
+          - name: WORDPRESS_EMAIL
+            valueFrom:
+              secretKeyRef:
+                name: wordpress-secret
+                key: email-wp
+        envFrom:
```
[the remark this answers is not in the document]
Comment (larabjj) [reply pablo_lara/session-01/exercise-01/wordpress-canary-deployment.yaml]: For nothing, you're right.

File: pablo_lara/session-01/exercise-01/wordpress-secret.yaml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  #WORDPRESS_PASSWORD: bjjtimes | base64
+  password-wp: YmpqdGltZXM=
+
+  #WORDPRESS_EMAIL: user@example.com | base64
+  email-wp: dXNlckBleGFtcGxlLmNvbQ==
+
+  #MARIADB_HOST: mariadb-deployment | base64
+  mariadb-hostname: bWFyaWFkYi1kZXBsb3ltZW50
+
+  #MARIADB_DATABASE_PASSWORD: bitnami | base64
```
[the remark this answers is not in the document]
Comment (larabjj) [reply pablo_lara/session-01/exercise-01/wordpress-secret.yaml]: Related to WP --> WORDPRESS_DATABASE_PASSWORD

File: pablo_lara/session-01/exercise-01/wordpress-secret.yaml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  #WORDPRESS_USERNAME: JiuJitsu | base64
+  username-wp: Sml1Sml0c3U=
+
+  #WORDPRESS_PASSWORD: bjjtimes | base64
+  password-wp: YmpqdGltZXM=
+
+  #WORDPRESS_EMAIL: user@example.com | base64
+  email-wp: dXNlckBleGFtcGxlLmNvbQ==
+
+  #MARIADB_HOST: mariadb-deployment | base64
```
[the remark this answers is not in the document]
Comment (larabjj) [reply pablo_lara/session-01/exercise-01/wordpress-secret.yaml]: you're right !

File: pablo_lara/session-01/exercise-01/mariadb-deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    metadata:
+      labels:
+        app: wordpress
+        tier: backend
+    spec:
+      containers:
+      - image: bitnami/mariadb:latest
+        livenessProbe:
+          exec:
+            command:
```
[the remark this answers is not in the document]
Comment (juan131) [reply pablo_lara/session-01/exercise-01/mariadb-deployment.yaml]: much better

File: pablo_lara/session-01/exercise-01/wordpress-svc.yaml [hunk 11]
```diff
@@ -7,9 +7,11 @@ metadata:
     app: wordpress
     tier: frontend
 spec:
-  type: NodePort
+  type: LoadBalancer
   ports:
     - port: 80
+      targetPort: http-wp
+      nodePort: 32500
```
Comment (juan131) [inline pablo_lara/session-01/exercise-01/wordpress-svc.yaml]: Why this weird **nodePort**?

Comment (larabjj) [reply pablo_lara/session-01/exercise-01/wordpress-svc.yaml]: I' fixed that nodePort to I can go kubernetes:32500 always. Shouldn't I do it ?

**Pull request merged**